backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -82,6 +82,28 @@ public function putTransaction($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function sellCards()
+    {
+        //add selling car to listing
+        $user = auth()->user();
+        $data = json_decode(Request::getContent(), true);
+        foreach ($data as $cardId) {
+            $card = Card::find($cardId);
+
+            $listing = new Listing();
+            $listing->card_id = $cardId;
+            $listing->user_id = $user->id;
+            $listing->price = $this->estimateValue($cardId);
+            $listing->save();
+
+            //remove selling card from user's cards
```

Comment:
The card should not be removed from the user until the Listing has found a buyer, at which point the card will be removed from the old user and added to the new user instantaneously.